### PR TITLE
AEA-1188 Make MultiAgentManager issue certificates

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Make the `MultiAgentManager` issue certificates for each agent just before they are run.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/cli/issue_certificates.py [hunk 1]
```diff
@@ -32,10 +32,10 @@
 from aea.cli.utils.package_utils import get_package_path_unified
 from aea.configurations.base import ConnectionConfig, PublicId
 from aea.configurations.constants import CONNECTION
+from aea.crypto.certificates import CertRequest
 from aea.crypto.helpers import make_certificate
 from aea.crypto.registries import crypto_registry
 from aea.exceptions import enforce
-from aea.helpers.base import CertRequest
```
Comment (marcofavorito) [inline aea/cli/issue_certificates.py:38]: moving `CertRequest` to `aea.crypto` was not necessary to make this PR working. Nevertheless, I think it is a better place than `aea.helpers.base`.

File: docs/cli-vs-programmatic-aeas.md [hunk 2]
```diff
@@ -196,14 +196,12 @@ def run():
...
-            "not_before": "2021-01-01",
-            "public_key": "fetchai",
-            "save_path": cert_path,
-        },
+        identifier="acn",
+        ledger_id=FetchAICrypto.identifier,
+        not_after="2022-01-01",
+        not_before="2021-01-01",
+        public_key="fetchai",
+        save_path=cert_path,
```
Comment (marcofavorito) [inline docs/cli-vs-programmatic-aeas.md:204]: not related to this PR

File: docs/multi-agent-manager.md [hunk 3]
```diff
@@ -33,29 +33,43 @@ manager.add_project(weather_station_id)
...
+    "author": "fetchai",
+    "version": "0.13.0",
+    "type": "connection",
+    "cert_requests": [{
+      "identifier": "acn",
+      "ledger_id": "fetchai",
+      "not_after": '2022-01-01',
+      "not_before": '2021-01-01',
+      "public_key": "fetchai",
+      "save_path": "fetchai/weather_station/.certs/conn_cert.txt"
```
Comment (marcofavorito) [inline docs/multi-agent-manager.md:69]: this allows us to call the `issue_certificates_` function from the root directory of the agent manager.

File: tests/test_docs/test_multiagent_manager.py [hunk 4]
```diff
@@ -43,7 +41,6 @@ def _import_module_mock(arg):
     return import_module(arg)
 
 
-@pytest.mark.skip
```
Comment (marcofavorito) [inline tests/test_docs/test_multiagent_manager.py:46]: now it should work

File: docs/multi-agent-manager.md [hunk 5]
```diff
@@ -67,21 +81,36 @@ component_overrides = {
...
+import time
+
 manager.start_agent(weather_station_id.name)
+
 # wait for ~10 seconds for peer node to go live
+time.sleep(10.0)
+
 manager.start_agent(weather_client_id.name)
+
+time.sleep(5.0)
```
Comment (marcofavorito) [inline docs/multi-agent-manager.md:113]: looks like this solves the test issues @DavidMinarsch

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
LGTM, but would be nice to find a way to get rid of cli parts in manager

**Pull request merged**